src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -1,88 +1,61 @@
-import { checkedToSats, toSats } from "@domain/bitcoin"
-import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
-import { defaultTimeToExpiryInSeconds } from "@domain/bitcoin/lightning/invoice-expiration"
-import { DealerPriceServiceError } from "@domain/dealer-price"
-import { checkedtoCents } from "@domain/fiat"
 import { RateLimitConfig } from "@domain/rate-limit"
 import { RateLimiterExceededError } from "@domain/rate-limit/errors"
-import { WalletInvoiceFactory } from "@domain/wallet-invoices/wallet-invoice-factory"
 import { checkedToWalletId } from "@domain/wallets"
-import { WalletCurrency } from "@domain/shared"
-import { DealerPriceService } from "@services/dealer-price"
+import { NewDealerPriceService } from "@services/dealer-price"
 import { LndService } from "@services/lnd"
 import { WalletInvoicesRepository, WalletsRepository } from "@services/mongoose"
 import { consumeLimiter } from "@services/rate-limit"
+import { WalletInvoiceBuilder } from "@domain/wallet-invoices/wallet-invoice-builder"
 
 export const addInvoiceForSelf = async ({
   walletId,
   amount,
   memo = "",
 }: AddInvoiceForSelfArgs): Promise<LnInvoice | ApplicationError> => {
-  const walletIdChecked = checkedToWalletId(walletId)
-  if (walletIdChecked instanceof Error) return walletIdChecked
-
-  const wallets = WalletsRepository()
-  const wallet = await wallets.findById(walletIdChecked)
-  if (wallet instanceof Error) return wallet
-
-  const limitOk = await checkSelfWalletIdRateLimits(wallet.accountId)
-  if (limitOk instanceof Error) return limitOk
-
-  let addInvoiceWithAmount: (
-    args: AddInvoiceArgs,
-  ) => Promise<LnInvoice | ApplicationError>
-
-  switch (wallet.currency) {
-    case WalletCurrency.Btc:
-      addInvoiceWithAmount = addInvoiceSatsDenomiation
-      break
-    case WalletCurrency.Usd:
-      addInvoiceWithAmount = addInvoiceFiatDenomiation
-      break
+  const limitCheckFn = checkSelfWalletIdRateLimits
+  const buildWIBWithAmountFn = ({
+    walletInvoiceBuilder,
+    recipientWalletDescriptor,
+  }: BuildWIBWithAmountFnArgs) => {
+    return Promise.resolve(
```

Comment:
if the method does not return a promise why do you want to force the return of a promise?

Reply:
Changed: https://github.com/GaloyMoney/galoy/pull/1241/commits/74a3cf3088cd95f38d8092525cd89b3c6e4f737f